components/dashboard/src/projects/project-context.tsx

```diff
@@ -66,9 +77,24 @@ export function useCurrentProject(): Project | undefined {
 
             // Find project matching with slug, otherwise with name
             const project = projects.find((p) => Project.slug(p) === slugs.projectSlug);
+            if (!project && teams) {
+                // check other orgs
+                for (const t of teams) {
+                    if (t.id === team?.id) {
+                        continue;
+                    }
+                    const projects = await listAllProjects({ teamId: t.id });
+                    const project = projects.find((p) => Project.slug(p) === slugs.projectSlug);
+                    if (project) {
+                        // redirect to the other org
+                        history.push(location.pathname + "?org=" + t.id);
```

Comment:
If you're on a project page, and try to switch orgs, because the org selector menu uses the current page as the link, you end up just getting redirected back to the project you were on and the org you were on. 

We can fix this by changing those org selector links to just link to `/workspaces?org=${orgId}` and not try to maintain the current page. Conversation [here](https://gitpod.slack.com/archives/C02EN94AEPL/p1675800615601449).

Reply:
This worked fine on the first attempt, redirecting you to the projects (list) of the second org, but after that I now always run into the same issue as described above.

> If you're on a project page, and try to switch orgs, because the org selector menu uses the current page as the link, you end up just getting redirected back to the project you were on and the org you were on.

Nice catch, @selfcontained! ⚾

Reply:
So changing the org selector should always navigate to the workspaces list?

Reply:
@svenefftinge Redirecting to a default page within the new org selected changes the context or path for the whole page, see https://github.com/gitpod-io/gitpod/issues/16243. ✔️ 

Keeping the user on the project page but updating the navigation context can create a slight confusion on whether the project belongs to the new org selected, as we've moved away from the breadcrumb pattern. ❌
 
We could introduce a breadcrumb line above the page header (project title) but still, updating only the navigation to a new context does not seem much helpful, right? 💡